modules/loader/src/main/java/com/opengamma/strata/loader/csv/FxSingleTradeCsvPlugin.java

```diff
@@ -178,19 +180,18 @@ static FxSingle parseFxSingle(CsvRow row, String prefix) {
 
   //-------------------------------------------------------------------------
   @Override
-  public List<String> headers(List<FxSingleTrade> trades) {
+  public Set<String> headers(List<FxSingleTrade> trades) {
     return HEADERS;
   }
 
   @Override
   public void writeCsv(CsvRowOutputWithHeaders csv, FxSingleTrade trade) {
     csv.writeCell(TRADE_TYPE_FIELD, "FxSingle");
-    writeProduct(csv, "", trade.getProduct());
+    writeFxSingle(csv, "", trade.getProduct());
     csv.writeNewLine();
   }
 
-  // writes the product to CSV
-  void writeProduct(CsvRowOutputWithHeaders csv, String prefix, FxSingle product) {
+  protected void writeFxSingle(CsvOutput.CsvRowOutputWithHeaders csv, String prefix, FxSingle product) {
```

Comment:
Does this really need to be `protected`? I doubt it. Either this class should be `final` and this method should be `private`, or this method should be package-scoped,